addons/osfstorage/tests/test_models.py

```diff
@@ -578,6 +579,37 @@ def test_after_fork_copies_versions(self):
         assert_equal(list(cloned_record.versions.all()), list(record.versions.all()))
         assert_true(fork_node_settings.root_node)
 
+    def test_fork_reverts_to_using_user_storage_default(self):
+        user = UserFactory()
+        user2 = UserFactory()
+        us = RegionFactory()
+        canada = RegionFactory()
+
+        user_settings = user.get_addon('osfstorage')
+        user_settings.default_region = us
+        user_settings.save()
+        user2_settings = user2.get_addon('osfstorage')
+        user2_settings.default_region = canada
+        user2_settings.save()
+
+        project = ProjectFactory(creator=user, is_public=True)
+        child = NodeFactory(parent=project, creator=user, is_public=True)
+        child_settings = child.get_addon('osfstorage')
+        child_settings.region_id = canada.id
+        child_settings.save()
+
+        fork = project.fork_node(Auth(user))
+        child_fork = models.Node.objects.get_children(fork).first()
+        assert fork.get_addon('osfstorage').region_id == us.id
+        assert fork.get_addon('osfstorage').user_settings == user.get_addon('osfstorage')
+        assert child_fork.get_addon('osfstorage').region_id == us.id
```

Comment:
US Project and Canada component are both forked to be a US Project and US component because the user's default region is the US